### ISSUE TITLE
salt-cloud preflight commands use system __opts__ not passed opts

### ISSUE THREAD

**Issue by vernondcole** [saltstack/salt#43497]
### Description of Issue/Question

The new preflight command uses `__opts__` rather than the 'opts' argument passed to the bootstrap method.  This can cause errors if bootstrap is called with a different configuration.

[Yes, I know that no one but me ever wants to do that.  My use case is weird.]

I have the single-line fix prepared and will submit a PR in a matter of moments after submitting this issue.

### Steps to Reproduce Issue

call bootstrap from a system where `__opts__` does not have a 'profiles' key defined.

```
  File "/ --- my source tree  ---  modules/vagrant.py", line 335, in _start
    ret = __utils__['cloud.bootstrap'](vm_, fudged_opts)
  File "/projects/SaltStack/salt/utils/cloud.py", line 460, in bootstrap
    'preflight_cmds', vm_, __opts__, default=[]
  File "/projects/SaltStack/salt/config/__init__.py", line 3106, in get_cloud_config_value
    if name in opts['profiles'][vm_['profile']]:
KeyError: 'profiles'
```

### Versions Report
Salt Version:
           Salt: 2017.7.0-570-g05fe5be
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: 2.0.2
      gitpython: 2.1.5
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.10 (default, Feb  7 2017, 00:08:15)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2
 
System Versions:
           dist:   
         locale: UTF-8
        machine: x86_64
        release: 16.7.0
         system: Darwin
        version: 10.12.6 x86_64

**Comment by rallytime** [comment]
Oh, you're right. Good catch. Thank you for filing this issue and for submitting the fix. This indeed should be using the passed in `opts` and not `__opts__`.

**Comment by rallytime** [comment]
Closed via #43498
